From: Joachim Bauch <jojo@struktur.de>
Date: Tue, 7 Jul 2015 18:28:46 +0200
Subject: safe_browsing: disable reporting of safebrowsing override

Disables reporting of the safebrowsing override, i.e. the report sent
if a user decides to visit a page that was flagged as "insecure".
This prevents trk:148 (phishing) and trk:149 (malware).
---
 .../safe_browsing/client_side_detection_service.cc   | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/chrome/browser/safe_browsing/client_side_detection_service.cc b/chrome/browser/safe_browsing/client_side_detection_service.cc
--- a/chrome/browser/safe_browsing/client_side_detection_service.cc
+++ b/chrome/browser/safe_browsing/client_side_detection_service.cc
@@ -61,11 +61,13 @@ enum MalwareReportTypes {
   REPORT_RESULT_MAX
 };
 
+#if 0
 void UpdateEnumUMAHistogram(MalwareReportTypes report_type) {
   DCHECK(report_type >= 0 && report_type < REPORT_RESULT_MAX);
   UMA_HISTOGRAM_ENUMERATION("SBClientMalware.SentReports", report_type,
                             REPORT_RESULT_MAX);
 }
+#endif
 
 }  // namespace
 
@@ -277,6 +279,10 @@ void ClientSideDetectionService::StartClientReportPhishingRequest(
     return;
   }
 
+#if 1
+  if (!callback.is_null())
+    callback.Run(GURL(request->url()), false);
+#else
   // Fill in metadata about which model we used.
   if (is_extended_reporting) {
     request->set_model_filename(model_loader_extended_->name());
@@ -359,6 +365,7 @@ void ClientSideDetectionService::StartClientReportPhishingRequest(
 
   // Record that we made a request
   phishing_report_times_.push(base::Time::Now());
+#endif
 }
 
 void ClientSideDetectionService::StartClientReportMalwareRequest(
@@ -373,6 +380,10 @@ void ClientSideDetectionService::StartClientReportMalwareRequest(
     return;
   }
 
+#if 1
+  if (!callback.is_null())
+    callback.Run(GURL(request->url()), GURL(request->url()), false);
+#else
   std::string request_data;
   if (!request->SerializeToString(&request_data)) {
     UpdateEnumUMAHistogram(REPORT_FAILED_SERIALIZATION);
@@ -442,6 +453,7 @@ void ClientSideDetectionService::StartClientReportMalwareRequest(
 
   // Record that we made a malware request
   malware_report_times_.push(base::Time::Now());
+#endif
 }
 
 void ClientSideDetectionService::HandlePhishingVerdict(
-- 
2.17.1

